examples/transpose/transpose_dist_object.cpp

```diff
@@ -0,0 +1,380 @@
+/// NUMA domian in a SMP machine.) Each locality constructs an instance of
+/// dist_object<T>, where a value of type T represents the value of this
+/// this locality's instance value. Once dist_object<T> is conctructed, it
+/// has a universal name which can be used on any locality in the given
+/// localities to locate the resident instance.
+
+#include <hpx/hpx.hpp>
+#include <hpx/hpx_init.hpp>
+
+#include <hpx/parallel/algorithms/for_each.hpp>
+#include <hpx/parallel/algorithms/transform_reduce.hpp>
+
+#include <boost/range/irange.hpp>
+
+#include <hpx/include/async.hpp>
+#include <hpx/include/dataflow.hpp>
+#include <hpx/lcos/barrier.hpp>
+#include <hpx/lcos/dist_object.hpp>
+#include <hpx/lcos/when_all.hpp>
+
+#include <algorithm>
+#include <cassert>
+#include <iostream>
+#include <string>
+#include <vector>
+
+bool verbose = false;    // command line argument
+
+typedef double* sub_block;
+
+#define COL_SHIFT 1000.00    // Constant to shift column index
+#define ROW_SHIFT 0.01       // Constant to shift row index
+
+// Register type for template components
+REGISTER_DIST_OBJECT_PART(double);
+using myVectorDouble = std::vector<double>;
+REGISTER_DIST_OBJECT_PART(myVectorDouble);
+
+using hpx::lcos::dist_object;
+
+///////////////////////////////////////////////////////////////////////////////
+// transpose matrix when the target matrix is in a remote node
+void transpose(hpx::future<std::vector<double>> Af, std::uint64_t A_offset,
+    dist_object<std::vector<double>>& B_temp, std::uint64_t B_offset,
+    std::uint64_t block_size, std::uint64_t block_order,
+    std::uint64_t tile_size);
+
+///////////////////////////////////////////////////////////////////////////////
+// transpose matrix when the target and destination matrix are in a same node
+void transpose_local(dist_object<std::vector<double>>& A_temp,
+    std::uint64_t A_offset, dist_object<std::vector<double>>& B_temp,
+    std::uint64_t B_offset, std::uint64_t block_size, std::uint64_t block_order,
+    std::uint64_t tile_size);
+
+double test_results(std::uint64_t order, std::uint64_t block_order,
+    std::vector<dist_object<std::vector<double>>>& trans,
+    std::uint64_t blocks_start, std::uint64_t blocks_end);
+
+///////////////////////////////////////////////////////////////////////////////
+void run_matrix_transposition(boost::program_options::variables_map& vm)
+{
+    hpx::id_type here = hpx::find_here();
+    bool root = here == hpx::find_root_locality();
+
+    std::uint64_t num_localities = hpx::get_num_localities().get();
+
+    std::uint64_t order = vm["matrix_size"].as<std::uint64_t>();
+    std::uint64_t iterations = vm["iterations"].as<std::uint64_t>();
+    std::uint64_t num_local_blocks = vm["num_blocks"].as<std::uint64_t>();
+    std::uint64_t tile_size = order;
+
+    if (vm.count("tile_size"))
+        tile_size = vm["tile_size"].as<std::uint64_t>();
+
+    verbose = vm.count("verbose") ? true : false;
```

Comment:
`<boolean expression> ? true : false` is always redundant ;) You might want to drop everything after `?` or do an explicit `vm.count("verbose") > 0`.

Reply:
@msimberg just writing `verbose = vm.count("verbose");` generates a stupid MSVC warning, so doing an explicit `vm.count("verbose") > 0` seems to be best.

Reply:
@msimberg @hkaiser please check if this looks good for you. Thanks!
https://github.com/STEllAR-GROUP/hpx/blob/b1c8f32ae0627852cf02f96013a70b443f03a066/examples/transpose/transpose_distributed_object.cpp#L87